### PR TITLE
chore (react-router-guard): remove redundant types

### PR DESCRIPTION
**Pull request by 43081j**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

---

ships its own types since 2.3.0:

https://unpkg.com/browse/react-router-guard@2.3.0/dist/index.d.ts

### CODE REVIEW

**Comment by 43081j** [comment]
blocked until a DT maintainer can tell me what im meant to do with the version here....

the DT version is (incorrectly?) 2.3.3 which doesn't even exist at the source package. it has been bumped beyond the package it provides types for.

so i can't mark the `asOfVersion` as the actual version which started shipping types, as that was 2.3.0... which is less than 2.3.3

i can't trick it either by just picking a higher shipped version, because there isn't one. the source package only goes to 2.3.1.

bumping this DT package to a non-existent version wasn't a good idea..

**Comment by 43081j** [comment]
No worries that's exactly what I did in the end

These two packages have been prep for a larger package being removed. I agree I doubt these are used anymore

**Comment by 43081j** [comment]
@sanders any chance you can review this one since i doubt the maintainer will turn up?

i can push my other PR then

**Review by sandersn — APPROVED** [review]

**Pull request merged**
